### PR TITLE
Lint fixes and clang analyze

### PR DESCRIPTION
**Pull request by theopolis**
Mostly fixing lint problems and clang analyze output from the 3.6 compiler on OS X.

### CODE REVIEW

File: tools/provision/ubuntu.sh [hunk 1]
```diff
@@ -54,12 +54,11 @@ function main_ubuntu() {
     sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-4.8 150 \
       --slave /usr/bin/g++ g++ /usr/bin/g++-4.8
     install_boost
-    install_cmake
   else
-    package cmake
     package libboost1.55-all-dev
   fi
 
+  install_cmake
```
Comment (theopolis) [inline tools/provision/ubuntu.sh]: So maybe this doesn't work on ubuntu14?

**Comment by theopolis** [comment]
ok to test

**Comment by theopolis** [comment]
Notes:

[Some](https://bugzilla.redhat.com/show_bug.cgi?id=1090716) say the error we're receiving installing CMake 3.2.1 on ubuntu14 is an error with Clang 3.4.0:
```
CMake has bootstrapped.  Now run gmake.
make[1]: Entering directory `/home/osquery/jenkins/workspace/osqueryPullRequestBuild/TargetSystem/ubuntu14/.sources/cmake-3.2.1'
make[2]: Entering directory `/home/osquery/jenkins/workspace/osqueryPullRequestBuild/TargetSystem/ubuntu14/.sources/cmake-3.2.1'
make[3]: Entering directory `/home/osquery/jenkins/workspace/osqueryPullRequestBuild/TargetSystem/ubuntu14/.sources/cmake-3.2.1'
Scanning dependencies of target cmIML_test
make[3]: Leaving directory `/home/osquery/jenkins/workspace/osqueryPullRequestBuild/TargetSystem/ubuntu14/.sources/cmake-3.2.1'
make[3]: Entering directory `/home/osquery/jenkins/workspace/osqueryPullRequestBuild/TargetSystem/ubuntu14/.sources/cmake-3.2.1'
[  0%] Building C object Utilities/KWIML/test/CMakeFiles/cmIML_test.dir/test.c.o
[  0%] Building C object Utilities/KWIML/test/CMakeFiles/cmIML_test.dir/test_ABI_C.c.o
[  1%] Building C object Utilities/KWIML/test/CMakeFiles/cmIML_test.dir/test_INT_C.c.o
[  1%] Building C object Utilities/KWIML/test/CMakeFiles/cmIML_test.dir/test_include_C.c.o
[  1%] Building CXX object Utilities/KWIML/test/CMakeFiles/cmIML_test.dir/test_ABI_CXX.cxx.o
[  1%] Building CXX object Utilities/KWIML/test/CMakeFiles/cmIML_test.dir/test_INT_CXX.cxx.o
[  1%] Building CXX object Utilities/KWIML/test/CMakeFiles/cmIML_test.dir/test_include_CXX.cxx.o
In file included from /home/osquery/jenkins/workspace/osqueryPullRequestBuild/TargetSystem/ubuntu14/.sources/cmake-3.2.1/Utilities/KWIML/test/test_include_CXX.cxx:12:
In file included from /usr/bin/../lib/gcc/x86_64-linux-gnu/4.8/../../../../include/c++/4.8/string:52:
In file included from /usr/bin/../lib/gcc/x86_64-linux-gnu/4.8/../../../../include/c++/4.8/bits/basic_string.h:2815:
In file included from /usr/bin/../lib/gcc/x86_64-linux-gnu/4.8/../../../../include/c++/4.8/ext/string_conversions.h:43:
/usr/bin/../lib/gcc/x86_64-linux-gnu/4.8/../../../../include/c++/4.8/cstdio:120:11: error: no member named 'gets' in the global namespace
  using ::gets;
        ~~^
1 error generated.
```

**Comment by theopolis** [comment]
Not a clang thing :/

File: tools/provision/ubuntu.sh [hunk 2]
```diff
@@ -32,8 +32,20 @@ function main_ubuntu() {
...
   package devscripts
   package debhelper
-  package clang-3.4
-  package clang-format-3.4
+
+  if [[ $DISTRO = "precise" ]]; then
+    package clang-3.4
+    package clang-format-3.4
+  else
+    package clang-3.5
```
Comment (theopolis) [inline tools/provision/ubuntu.sh]: @marpaia installing 3.5 on trusty since it's available by default :)

**Comment by theopolis** [comment]
ok to test

**Comment by marpaia** [comment]
Looks great!

**Pull request merged**